src/apply.jl

```diff
@@ -87,52 +86,124 @@ end
 function ITensors.apply(
   o⃗::Vector{ITensor},
   ψ::AbstractITensorNetwork;
-  cutoff,
-  maxdim=typemax(Int),
   normalize=false,
   ortho=false,
-  kwargs...,
+  apply_kwargs...,
 )
   o⃗ψ = ψ
   for oᵢ in o⃗
-    o⃗ψ = apply(oᵢ, o⃗ψ; cutoff, maxdim, normalize, ortho)
+    o⃗ψ = apply(oᵢ, o⃗ψ; normalize, ortho, apply_kwargs...)
   end
   return o⃗ψ
 end
 
 function ITensors.apply(
-  o⃗::Scaled,
-  ψ::AbstractITensorNetwork;
-  cutoff,
-  maxdim,
-  normalize=false,
-  ortho=false,
-  kwargs...,
+  o⃗::Scaled, ψ::AbstractITensorNetwork; normalize=false, ortho=false, apply_kwargs...
 )
   return maybe_real(Ops.coefficient(o⃗)) *
-         apply(Ops.argument(o⃗), ψ; cutoff, maxdim, normalize, ortho, kwargs...)
+         apply(Ops.argument(o⃗), ψ; cutoff, maxdim, normalize, ortho, apply_kwargs...)
 end
 
 function ITensors.apply(
-  o⃗::Prod,
-  ψ::AbstractITensorNetwork;
-  cutoff,
-  maxdim,
-  normalize=false,
-  ortho=false,
-  kwargs...,
+  o⃗::Prod, ψ::AbstractITensorNetwork; normalize=false, ortho=false, apply_kwargs...
 )
   o⃗ψ = ψ
   for oᵢ in o⃗
-    o⃗ψ = apply(oᵢ, o⃗ψ; cutoff, maxdim, normalize, ortho, kwargs...)
+    o⃗ψ = apply(oᵢ, o⃗ψ; normalize, ortho, apply_kwargs...)
   end
   return o⃗ψ
 end
 
 function ITensors.apply(
-  o::Op, ψ::AbstractITensorNetwork; cutoff, maxdim, normalize=false, ortho=false, kwargs...
+  o::Op, ψ::AbstractITensorNetwork; normalize=false, ortho=false, apply_kwargs...
 )
-  return apply(ITensor(o, siteinds(ψ)), ψ; cutoff, maxdim, normalize, ortho, kwargs...)
+  return apply(ITensor(o, siteinds(ψ)), ψ; normalize, ortho, apply_kwargs...)
+end
+
+#In the future we will try to unify this into apply() above but currently leave it mostly as a separate function
+"""Apply() function for an ITN in the Vidal Gauge. Hence the bond tensors are required.
+Gate does not necessarily need to be passed. Can supply an edge to do an identity update instead. Uses Simple Update procedure assuming gate is two-site"""
+function ITensors.apply(
+  o::Union{ITensor,NamedEdge},
+  ψ::AbstractITensorNetwork,
+  bond_tensors::DataGraph;
+  normalize=false,
+  apply_kwargs...,
+)
+  ψ = copy(ψ)
+  bond_tensors = copy(bond_tensors)
+  v⃗ = typeof(o) == ITensor ? neighbor_vertices(ψ, o) : [src(o), dst(o)]
+  if length(v⃗) == 2
+    e = NamedEdge(v⃗[1] => v⃗[2])
+    ψv1, ψv2 = copy(ψ[src(e)]), copy(ψ[dst(e)])
+    e_ind = commonind(ψv1, ψv2)
+
+    for vn in neighbors(ψ, src(e))
+      if (vn != dst(e))
+        ψv1 = noprime(ψv1 * bond_tensors[vn => src(e)])
+      end
+    end
+
+    for vn in neighbors(ψ, dst(e))
+      if (vn != src(e))
+        ψv2 = noprime(ψv2 * bond_tensors[vn => dst(e)])
+      end
+    end
+
+    if typeof(o) == ITensor
+      G1, G2 = factorize(o, Index[commonind(ψv1, o), commonind(ψv1, o)']; cutoff=1e-16)
+      ψv1 = noprime(ψv1 * G1)
+      ψv2 = noprime(ψv2 * G2)
+    end
```

Comment:
Same comment as above, I would split this off into a separate function like:
```julia
function _contract_factorized_gate(o::ITensor, ψv1, ψv2)
  G1, G2 = factorize(o, Index[commonind(ψv1, o), commonind(ψv1, o)']; cutoff=1e-16)
  ψv1 = noprime(ψv1 * G1)
  ψv2 = noprime(ψv2 * G2)
  return ψv1, ψv2
end

function _contract_factorized_gate(o::AbstractEdge, ψv1, ψv2)
  return ψv1, ψv2
end
```
I think splitting off small functions like this helps "self-document" the code and helps to identify patterns for how the code could be generalized (for example, for other gate types, like gates that are MPOs or TTNOs).

Reply:
Yeah I can see that that is a neater way to write the code! Thanks